### PR TITLE
Tier 2 - CEPH-83575085

### PR DESCRIPTION
**Pull request by hkadam134**
Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-M0A297/

TC : CEPH-83575085 - Create an export on the NFS cluster and mount it using all supported NFS versions(v4,4,1,4.2) using the backend servers as well as the ingress and perform IO along with failover.

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>
<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by neha-gangadhar** [comment]
@hkadam134 can you please add the suite file to the PR so that we dont miss the executions of the automated tests.

**Comment by hkadam134** [comment]
> @hkadam134 can you please add the suite file to the PR so that we dont miss the executions of the automated tests.

4 ppl are working on the same suites and to avoid conflicts I intentionally didn't add, will take care of adding the suites before closing the ticket.

File: tests/cephfs/cephfs_utilsV1.py [hunk 1]
```diff
@@ -3097,3 +3098,30 @@ def nfs_mount_and_io(self, nfs_client, nfs_server, export_name, mount_path):
             f"{mount_path}/{dir_name}",
             long_running=True,
         )
+
+    def create_directory(self, client, directory):
```
Comment (AmarnatReddy) [inline tests/cephfs/cephfs_utilsV1.py:3102]: can we directly run mkdir in line 3125

Comment (hkadam134) [reply tests/cephfs/cephfs_utilsV1.py:3102]: wanted to reduce the creation of directories in the test scripts by adding a common function. 
anything wrong with having this as a common function?

File: tests/cephfs/cephfs_nfs/test_all_nfs_vers_mount_and_perform_failover.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+def run(ceph_cluster, **kw):
+    """
+    CEPH-83575085 - Create an export on the NFS cluster and mount it using all
+                    supported NFS versions(4,1,4.2) using the backend servers as well as the ingress
+                    and perform IO as usual.
+    Test operation:
+        1. Create a nfs cluster
+        2. Create a nfs export
+        3. Create 2 exports
+        4. Mount 1st export with nfs version 4.1 and run IOs
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_nfs/test_all_nfs_vers_mount_and_perform_failover.py:24]: Can we also include version as 4 in the same test? We might be covering it other test. Just a thought, open for discussion.

File: tests/cephfs/cephfs_utilsV1.py (hunk 1 above)
Comment (neha-gangadhar) [reply tests/cephfs/cephfs_utilsV1.py:3102]: There is no harm in creating function. However the same could have been used in line 118. Also, other team members remembering a function exists for creating a directory is low as it is easier to fire up a single mkdir command when needed. Reusable function should have multiple lines and good chunk of function handled is the thought process.

**Comment by neha-gangadhar** [comment]
Lgtm with minor comments.

File: tests/cephfs/cephfs_utilsV1.py (hunk 1 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_utilsV1.py]: Updated

File: tests/cephfs/cephfs_nfs/test_all_nfs_vers_mount_and_perform_failover.py (hunk 2 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_nfs/test_all_nfs_vers_mount_and_perform_failover.py:24]: added version 4 also.

**Comment by hkadam134** [comment]
Latest Logs after addressing all the comments - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-JTUL1O/

**Review by neha-gangadhar — APPROVED** [review]
lgtm

**Pull request merged**